src/main/java/org/repodriller/scm/GitRepository.java

```diff
@@ -232,7 +232,11 @@ public Commit getCommit(String id) {
 				if(jgitCommit.getParentCount() > 1) merge = true;
 				Set<String> branches = getBranches(git, hash);
 				boolean isCommitInMainBranch = branches.contains(this.mainBranchName);
-				theCommit = new Commit(hash, author, committer, authorDate, authorTimeZone, committerDate, committerTimeZone, msg, parent, merge, branches, isCommitInMainBranch);
+				List<ChangeSet> changeSets = getChangeSets();
+				int totalCommits = changeSets.size();
+				long commitPosition = totalCommits - changeSets.stream().map((cs)->cs.getId()).collect(Collectors.toList()).indexOf(hash);
```

Comment:
Although this command is written in 1 line, it maps hashes through all commits every getCommit()